kernel/src/drivers/terminal/mod.rs

```diff
@@ -0,0 +1,316 @@
+use core::ops::Add;
+use core::fmt;
+use core::result::Result;
+use spin::Mutex;
+use drivers::vga::{self, VgaWriter};
+use color::Color;
+
+pub static WRITER: Mutex<VgaWriter> = Mutex::new(VgaWriter::new());
+pub static STDOUT: Mutex<TextArea<VgaWriter>> = Mutex::new(TextArea::new(&WRITER, Point::new(0, 0), Point::new(vga::RESOLUTION_X, vga::RESOLUTION_Y)));
```

Comment:
Also, the formatting here is *weird*. We should run rustfmt on this whole project tbh

Reply:
That's for another PR though. Let me add that as an issue.